rdflib/extras/infixowl.py

```diff
@@ -382,17 +386,71 @@ def __init__(self, identifier=None, graph=None):
             except Exception:  # pragma: no cover
                 pass  # pragma: no cover
 
+    def subject_node_closure(self, graph):
+        """
+        Take terms referencing this individual as a subject and
+        add them to the provided graph.
+        """
+        for fact in self.factoryGraph.triples((self.identifier, None, None)):
+            graph.add(fact)
+
+        return graph
+
+    def blank_node_closure(self, graph):
+        """
+        Take terms related to this individual using a blank node
+        closure and add them to the provided graph.
+        """
+        for fact in self.factoryGraph.triples((self.identifier, None, None)):
+            graph.add(fact)
+            if (
+                isinstance(fact[2], BNode)
+                and (fact[2], None, None) not in self.factoryGraph
+            ):
+                graph += self.factoryGraph.triples((fact[2], None, None))
+
+        return graph
```

Comment:
I wanted to add tests for these, but I'm actually not entirely sure how they should be used, in part because I'm not certain why they use `self.factoryGraph` instead of `self.graph`, should they use `self.graph` instead?

Reply:
I should have provided some context for this, a lacuna I now remedy. The origin of this particular addition is Will Waites’ work on “Beginnings of an Object Description Mapper” originally advertised on the [FuXI Google groups discussion](https://groups.google.com/g/fuxi-discussion/c/SO5IE-WqlOw/m/ODOBB6fCwBwJ). Although the OKFN no longer supports ORDF, the adapted infixowl referenced in Will’s post remains [available via PyPi](https://pypi.org/project/ordf/) and this is the origin of this particular change:
> 3. The Individual.serialize method changed to use the bnc method on the graph which gives the blank node closure of the individual rather than just those triples mentioning it as a subject. This is a method I would really like to see on the upstream rdflib graph. I am considering changing this further to replace this method with Individual.bnc() which just returns individual.graph.bnc((self.identifier, None, None)), I think this is cleaner looking and maybe serialize is confusingly named.

So, the original `bnc` parameter switch gets separated out into its own function and becomes `blank_node_closure` and  `serialize` is renamed to `subect_node_closure`.

I take your point about lack of tests, I'll address that and at the same time, find out whether `self.graph` and `self.factoryGraph` are freely interchangeable.

(fwiw, I have explored extending `infixowl` with the rest of Will’s ODM additions to get the behaviour/functionality as described in the [ORDF ODM doc](https://pythonhosted.org/ordf/odm.html), including the restitution of forward-chaining inferencing by updating and recruiting pychinko with its RETE implementation and - if I can manage to fix a particularly pernicious show-stopping ODM issue - may well offer an extended PR to this effect)

Reply:
> I take your point about lack of tests, I'll address that and at the same time, find out whether `self.graph` and `self.factoryGraph` are freely interchangeable.

Maybe the best option is to move that to a new PR? And just merge the rest? I think some of this would be good to get in, I can split off the rest of the changes and merge it if you want to continue with this.

What bothers me about `self.factoryGraph` is that it will shared between all instances, as far as I can tell.

Reply:
> What bothers me about `self.factoryGraph` is that it will shared between all instances, as far as I can tell.

The use of a class variable here is neither unintentional nor unconsidered - it was present in the [original commit](https://github.com/RDFLib/FuXi/blob/5978bcd9d08f759c51579922714b66197be7cefe/lib/Syntax/InfixOWL.py#L287).

There's some clue as to the anticipated usage evident in the docstrings where `Individual.factoryGraph` is successively [bound to different graphs](https://github.com/RDFLib/rdflib/blob/e103078be6c75b2f9a1dc515abffcdf6a674407d/rdflib/extras/infixowl.py#L1772).

I'll put in some more work on explicitly understanding its role but yes, it's certainly posing the exact problem you describe in the tests for Will’s ODM implementation.